File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
class Solution {
   	public static int[] solution(int[] arr, boolean[] flag) {
		int size = 0;
		int answersize=0;
		for (int i = 0; i < flag.length; i++) {
			size += flag[i] ? arr[i] * 2 : 0;
			answersize += flag[i] ? arr[i] * 2 : -arr[i];
		}
		int[] answertmp = new int[size];
		int k = 0;
		for (int i = 0; i < flag.length; i++) {
			if (flag[i]) {
				for (int j = 0; j < arr[i] * 2; j++) {
					answertmp[k] = arr[i];
					k++;
				}
			}else {
				k=Math.max(0,k-arr[i]);
			}
		}
		int[ ]answer= new int [answersize];
		
		for (int i = 0; i < answer.length; i++) {
			answer[i]=answertmp[i];
		}

		return answer;
	}
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
  let x = [];
  arr.forEach((e, i) => {
    if (flag[i]) {
      for (let i = 0; i < e * 2; i++) {
        x.push(e);
      }
    } else {
     x=x.slice(0, x.length - e );
    }
  });

  return x;
}
